tests/test_repository_lib.py

```diff
@@ -481,6 +481,184 @@ def test_generate_snapshot_metadata(self):
+    # Test normal case.
+    temporary_directory = tempfile.mkdtemp(dir=self.temporary_directory)
+    original_repository_path = os.path.join('repository_data',
+                                            'repository')
+    repository_directory = os.path.join(temporary_directory, 'repository')
+    shutil.copytree(original_repository_path, repository_directory)
+    metadata_directory = os.path.join(repository_directory,
+                                      repo_lib.METADATA_STAGED_DIRECTORY_NAME)
+    targets_directory = os.path.join(repository_directory, repo_lib.TARGETS_DIRECTORY_NAME)
+    targets_filename = os.path.join(metadata_directory,
+                                    repo_lib.TARGETS_FILENAME)
+    version = 1
+    expiration_date = '1985-10-21T13:20:00Z'
+
+    # Load a valid repository so that top-level roles exist in roledb and
+    # generate_snapshot_metadata() has roles to specify in snapshot metadata.
+    repository = repo_tool.Repository(repository_directory, metadata_directory,
+                                      targets_directory)
+
+    repository_junk = repo_tool.load_repository(repository_directory)
+
+    # For testing purposes, store an invalid metadata file in the metadata directory
+    # to verify that it isn't loaded by generate_snapshot_metadata().  Unknown
+    # metadata file extensions should be ignored.
+    invalid_metadata_file = os.path.join(metadata_directory, 'role_file.xml')
+    with open(invalid_metadata_file, 'w') as file_object:
+      file_object.write('bad extension on metadata file')
+
+    targets_filename = 'targets'
+
+    snapshot_metadata = \
+      repo_lib.generate_snapshot_metadata(metadata_directory, version,
+                                          expiration_date,
+                                          targets_filename,
+                                          consistent_snapshot=False,
+                                          use_hashes=False)
+    self.assertTrue(tuf.formats.SNAPSHOT_SCHEMA.matches(snapshot_metadata))
+
+    metadata_files_info_dict = snapshot_metadata['meta']
+    for metadata_filename in sorted(os.listdir(metadata_directory), reverse=True):
+
+      # In the metadata_directory, there are files with format:
+      # 1.root.json. The prefix number should be removed.
+      if metadata_filename[0].isdigit():
+        version_number, metadata_filename = metadata_filename.split('.', 1)
+
+      if metadata_filename.endswith('.json'):
+        if metadata_filename not in \
+            ['root.json', 'targets.json', 'timestamp.json', 'snapshot.json']:
+
+          # Check that hashes is not calculated but length is
+          self.assertTrue(
+            metadata_files_info_dict[metadata_filename].get('length') is not None)
+          self.assertTrue(
+            metadata_files_info_dict[metadata_filename].get('hashes') is None)
+
+
+
+  def test_generate_snapshot_metadata_without_hashes_and_length(self):
```

Comment:
Would it be possible to reuse the logic from the individual tests above?

Reply:
I rebased and addressed that comment.

Reply:
I rebased and addressed that comment.